src/viking.js

```diff
@@ -1,11 +1,111 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health,strength){
+        this.health=health;
+        this.strength=strength;
+    }
+    attack(){
+        return this.strength
+
+    }
+    receiveDamage(damage){
+        this.health -= damage
+
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name,health,strength){
+        super(health,strength)
+        this.name = name;
+
+    }
+    receiveDamage(damage){
+        this.health -= damage
+        if(this.health>0){
+            return ` ${this.name} has recived ${damage} points of damage`
+        }
+        else{
+            return `${this.name} has died in act of combat`
+        }
+    
+
+    }
+    battleCry(){
+        return 'Odin Owns You All!'
+    }
+}
+
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    receiveDamage(damage){
+       
+        this.health -= damage
```

Comment:
Aquí no necesitas volver a escribir "**this.health -= damage**", lo que hacemos es llamar al método original que ya lo hace:
**super.receiveDamage(damage);**